tests/hook_test.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import os
+import pytest
+import subprocess
+
+from git import Repo
+
+
+@pytest.fixture
+def tmp_repo(tmpdir):
+    # initialize new repository and change working directory
+    directory = str(tmpdir)
+    repo = Repo.init(directory)
+    os.chdir(directory)
+
+    # set up logging and trace
+    logging.basicConfig()
+    logging.root.setLevel(logging.INFO)
+    type(repo.git).GIT_PYTHON_TRACE = 'full'
+
+    # create initial commit
+    file_path = os.path.join(directory, 'initial-commit.txt')
+    subprocess.call(['touch', file_path])
+    repo.git.add('--all')
+    repo.git.commit('-m', '\'message\'', '--no-verify')
+
+    subprocess.call(['rdm', 'hooks'])
+
+    yield repo
+
+    # teardown
+    subprocess.call(['rm', '-rf', directory])
+
+
+def prepare_branch(tmp_repo, branch_name):
+    directory = os.getcwd()
+
+    tmp_repo.git.checkout('-b', branch_name)
+
+    file_path = os.path.join(directory, 'empty-file.txt')
+    subprocess.call(['touch', file_path])
+
+    tmp_repo.git.add('--all')
+    tmp_repo.git.commit('-m', 'Fix some issue')
+
+
+def show_commit_message():
+    return str(subprocess.check_output(
+        ['git', 'show', '-s', '--format=%B'],
+    ), 'utf-8')
+
+
+def test_single_issue(tmp_repo):
+    prepare_branch(tmp_repo, '10-sample-issue')
+
+    assert show_commit_message() == "Fix some issue\n\nIssue #10\n\n"

```

Comment:
IMO the new lines in these tests don't make the code easier to read, but do make the file longer.

Reply:
For this part I came up with a few options:

1. `.strip()` the string and remove the `\n` from the assertion
2. Rearrange it and use `assert "text" in message()`
3. Use multi-line string for readability to format it as it would appear in the commit message

Which would you suggest I do?

Reply:
I'm sorry; I actually meant the new lines in the tests themselves :).

I.e. use this:

```
def test_single_issue(tmp_repo):
    prepare_branch(tmp_repo, '10-sample-issue')
    assert show_commit_message() == "Fix some issue\n\nIssue #10\n\n"
```

I think the original assertions were probably better, since they were more precise.

Reply:
Oh okay, misunderstood that; I reverted the changes.